### PR TITLE
Stop 131

### PR DESCRIPTION
**Pull request by erikd234**
Hey Adam. give it a read and let me know if you see any potential bugs,

It works when I write a 1 to the GPIO pin 12 on the jetson sentry mode is on and instructions are sent,

Here is the design diagram

system_state is a new node I made that publishes state.

image

### CODE REVIEW

File: src/motor_instructions.py [hunk 1]
```diff
@@ -2,14 +2,8 @@
...
 from std_msgs.msg import Empty, Float32
-import imp
-import rospkg
-
-rospack = rospkg.RosPack()
-pck_path = rospack.get_path("bike_sentry")
-
-SystemState = imp.load_source("bike_sentry.SystemState", pck_path + "/scripts/SystemState.py")
-CDM = imp.load_source("module.name", pck_path + "/scripts/CameraDataManager.py")
+from CameraDataManager import CameraDataManager
```
Comment (afong3) [inline src/motor_instructions.py:4]: Nice, I assume you made CameraDataManager into a package? Was it hard?

File: src/motor_instructions.py [hunk 2]
```diff
@@ -1,11 +1,23 @@
...
 
+INPUT_PIN = 18 # pin 12 on the real board
+
+def pin_setup():
+    # Pin Setup:
+    GPIO.setmode(GPIO.BCM)  # BCM pin-numbering scheme from Raspberry Pi
+    GPIO.setup(INPUT_PIN, GPIO.IN, pull_up_down=GPIO.PUD_UP)  # set pin as an input pin
+
+def read_pin():
+    global PREV_VALUE
```
Comment (afong3) [inline src/motor_instructions.py]: I might be missing what this is doing but this doesn't seem to be referenced.

File: src/motor_instructions.py [hunk 3]
```diff
@@ -63,7 +50,11 @@ def send_instruction(self, instr):
 def init_subscribers():
     rospy.Subscriber("x_center", Float32, pan_callback)
     rospy.Subscriber("y_center", Float32, tilt_callback)
+    rospy.Subscriber("state", Int32, state_callback)
 
+def state_callback(state):
+    rospy.loginfo(state)
+    rospy.loginfo("Is working")
```
Comment (afong3) [inline src/motor_instructions.py]: The state doesn't seem to be affecting any of the code in this node. It looks like you fixed this in a more recent commit though.

**Review by afong3 — COMMENTED** [review]
Nice things look good. I had a couple comments about motor_instructions but it looked like you fixed them in a different commit.

File: src/motor_instructions.py (hunk 1 above)
Comment (erikd234) [reply src/motor_instructions.py:4]: Just to simplify everything I moved it to the same directory as our other code. Python will find it if its the same directory.

File: src/motor_instructions.py (hunk 2 above)
Comment (erikd234) [reply src/motor_instructions.py]: this is dead code see all commits instead of just this one.

**Review by afong3 — APPROVED** [review]
Looks good

**Pull request merged**